maltego_trx/oauth.py

```diff
@@ -25,24 +26,29 @@ def _rsa_decrypt(private_key_path=None, ciphertext=None):
         """
         RSA Decryption function, returns decrypted plaintext in b64 encoding
         """
-        dsize = SHA.digest_size
-        sentinel = Random.new().read(20 + dsize)
         ciphertext = base64.b64decode(ciphertext)
-        private_key = RSA.import_key(open(private_key_path).read())
-        cipher = PKCS1_v1_5.new(private_key)
-        plaintext = cipher.decrypt(ciphertext, sentinel).decode('utf8')
+
+        with open(private_key_path, "rb") as key_file:
+            private_key = serialization.load_pem_private_key(key_file.read(),
+                                                             password=None,
+                                                             backend=None)
+            plaintext = private_key.decrypt(ciphertext, padding.PKCS1v15())
+
         return plaintext
 
     @staticmethod
     def _aes_decrypt(key=None, ciphertext=None):
         """
         AES Decryption function, returns decrypted plaintext value
         """
-        unpad = lambda s: s[:-ord(s[len(s) - 1:])]
         key = base64.b64decode(key)
+        cipher = Cipher(algorithms.AES(key), modes.ECB(), backend=default_backend())
+        decryptor = cipher.decryptor();
         ciphertext = base64.b64decode(ciphertext)
-        cipher = AES.new(key, AES.MODE_ECB)
-        plaintext = unpad(cipher.decrypt(ciphertext)).decode('utf8')
+        padded_b64_plaintext = decryptor.update(ciphertext) + decryptor.finalize()
+
+        unpad = lambda data: data[:-ord(data[len(data) - 1:])]
```

Comment:
This looks like PKCS7 padding. Why not use `cryptography.hazmat.primitives.padding.PKCS7()`?

Reply:
Ack